### ISSUE TITLE
Extra commas appear in output

### ISSUE THREAD

**Issue by cgusb** [psf/black#2047]
**Black v20.8b1**

[Playground link](https://black.now.sh/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AHCAItdAD2IimZxl1N_WlwxQm1kLZbCMoi_i0Lnek4rPIdii3_lq57GaefN9mp9ivAGI-QsV0RZLOx26KWd2li4qHPvVFTo-L10K-wEhnty_fY6xZc6ejkFaYzXDPvGlEo9bJLMSoRZ6IUhKsHAbaBc69EkBa4ppbMINRNaoV4f5QILokh4jxkAS0Tin8sx1gAAAK09qxWV-z-4AAGnAcMDAACKI5KcscRn-wIAAAAABFla)

## Options

`--line-length=88`
`--safe`

## Input

```python
img_diff = (img[0:min(img.shape[0], img_to_subtract.shape[0]), 
                 0:min(img.shape[1], img_to_subtract.shape[1])] 
            - img_to_subtract[0:min(img.shape[0], 
                                    img_to_subtract.shape[0]), 
                              0:min(img.shape[1], 
                                    img_to_subtract.shape[1])])
```

## Output

```python
img_diff = (
    img[
        0 : min(img.shape[0], img_to_subtract.shape[0]),
        0 : min(img.shape[1], img_to_subtract.shape[1]),
    ]
    - img_to_subtract[
        0 : min(img.shape[0], img_to_subtract.shape[0]),
        0 : min(img.shape[1], img_to_subtract.shape[1]),
    ]
)

```

## Expected

```python
img_diff = (
    img[
        0 : min(img.shape[0], img_to_subtract.shape[0]),
        0 : min(img.shape[1], img_to_subtract.shape[1])  # Comma should not be placed at the end of this line like in Output
    ]
    - img_to_subtract[
        0 : min(img.shape[0], img_to_subtract.shape[0]),
        0 : min(img.shape[1], img_to_subtract.shape[1])  # Comma should not be placed at the end of this line like in Output
    ]
)

```

**Comment by ichard26** [comment]
Black used to remove the trailing comma if the expression fits in a single line, but this was changed by #826 and #1288. Now a trailing comma tells Black to always explode the expression. This change was made mostly for the cases where you _know_ a collection or whatever will grow in the future. Having it always exploded as one element per line reduces diff noise when adding elements. Before the "magic trailing comma" feature, you couldn't anticipate a collection's growth reliably since collections that fitted in one line were ruthlessly collapsed regardless of your intentions. One of Black's goals is reducing diff noise, so this was a good pragmatic change.

So no, this is not a bug, but an intended feature. The reason why you're filing this issue is probably since we say the following in the (outdated) style documentation:
> Unnecessary trailing commas are removed if an expression fits in one line. This makes it 1% more likely that your line won't exceed the allotted line length limit. Moreover, in this scenario, if you added another argument to your call, you'd probably fit it in the same line anyway. That doesn't make diffs any larger.

We missed that this paragraph became incorrect when the "magic trailing comma" feature was introduced. It was eventually fixed in commit 6b935a34d08c0b18d858e7da5093dc69a29c55f2, but that was after the stable documentation was released alongside `Black 20.8b1`. 

Anyway, [here's the documentation](https://github.com/psf/black/blob/master/docs/the_black_code_style.md#the-magic-trailing-comma) on the "magic trailing comma". Hopefully that helps and sorry for the possible confusion.

**Comment by cgusb** [comment]
@ichard26 Thank you for the explanation!
